core/bin/dust_api.rs

```diff
@@ -755,21 +766,29 @@ async fn runs_create(
 ) -> (StatusCode, Json<APIResponse>) {
     let mut credentials = payload.credentials.clone();
 
+    // Convert payload secrets vector to hash map to use them with {secrets.SECRET_NAME}.
+    let mut secrets_hashmap = HashMap::new();
```

Comment:
You can do all that in one line using an iter / collect and avoid creating intermediary variables which is more idiomatic in rust